### PR TITLE
Update to Error Prone 2.18.0

### PR DESCRIPTION
**Pull request by msridhar**
Fixed some minor Javadoc warnings reported by the new version.  Also, adds an `@Inject` annotation on the main `NullAway` constructor, for compatibility with future versions of Error Prone that load checkers via Guice.

### CODE REVIEW

File: build.gradle
Comment (lazaroclapp) [inline build.gradle:79]: Do we want to keep this around? I would think this is a one time config...

Comment (lazaroclapp) [inline build.gradle:79]:
```suggestion
```

**Review by lazaroclapp — COMMENTED** [review]
Mostly LGTM, just wondering if we want to keep the patch run commented config. I suspect we don't and it was a one time configuration?

File: build.gradle
Comment (msridhar) [reply build.gradle:79]: I thought it might be handy as an example in case we want to auto-patch in the future.  I can extend the comment to clarify its purpose and give full instructions (`-Werror` also needs to be removed).  Alternately I can just delete; WDYT?

**Comment by msridhar** [comment]
Also, from watching the Error Prone tracker, it seems another release may be coming very soon to fix some bugs (see [here](https://github.com/google/error-prone/issues/3653#issuecomment-1369013704)).  It may be best to just wait for that release and do that update in this PR.

File: build.gradle
Comment (lazaroclapp) [reply build.gradle:79]: Either works for me, but if we are leaving it as an example, then I'd indeed add a slightly longer comment/instructions and maybe just replace `MissingSummary` for `[CheckerName]`?

Comment (msridhar) [reply build.gradle]: Fixed in [1ff9f9f](https://github.com/uber/NullAway/pull/707/commits/1ff9f9f0cef0ac861afe23fb1e17f84c2bb8bd35)

**Comment by msridhar** [comment]
Switching to draft until next EP release comes out

**Review by lazaroclapp — APPROVED** [review]
LGTM. Let me know if you want me to change the default branch protection jobs and land this now or after the fix serialization changes.

**Comment by msridhar** [comment]
I have permission to change the job names. But let's just wait until after the other fixes go in; no urgency here

**Comment by msridhar** [comment]
@lazaroclapp I think this is ready for a re-review.  Once you approve, I will update the branch protection job names and then land.

**Review by lazaroclapp — APPROVED** [review]
LGTM! Either way I think we want to throughly test this next release internally before publishing, so making sure all is well with `@Inject` will be part of that. Please change branch protection stuff before landing! 🚀

**Pull request merged**
